src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
```

Comment:
The `index` variable is incremented at the beginning of the loop, but it is not used anywhere before the increment. This leads to skipping the first element of `dateArr` during conversion. It would be more appropriate to increment `index` in the place where it's really needed.